src/DatabaseManager.php

```diff
@@ -206,6 +206,24 @@ public function execute(string $query, array $params = []): int
     }
 
     /**
+     * @throws DatabaseException
+     */
+    public function executeTransaction(callable $transaction)
+    {
+        try {
+            $this->beginTransaction();
+            $result = $transaction($this);
+            $this->commit();
+
+            return $result;
+        } catch (DatabaseException $exception) {
```

Comment:
Should we wrap this in a TransactionException instead?

Reply:
Yeah, sounds good! I'll update it.

Reply:
Done.